config/v1/types.go

```diff
@@ -8,9 +8,18 @@ import (
 // ConfigMapFileReference references a config map in a specific namespace.
 // The namespace must be specified at the point of use.
 type ConfigMapFileReference struct {
+	// name is the metadata.name of the referenced config map
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]([-a-z0-9]*[a-z0-9])?$`
+	// +required
 	Name string `json:"name"`
-	// key allows pointing to a specific key/value inside of the configmap.  This is useful for logical file references.
-	Key string `json:"key,omitempty"`
+
+	// key is the key inside the referenced ConfigMap pointing to the custom logo file reference.
+	// +required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Key string `json:"key"`
```

Comment:
The changes made here have an impact to every type across OpenShift that uses this. Adding the validations that you've added are _all_ breaking changes. I would recommend reverting this.

My comments regarding the validations on these things were specifically for when you were not using this existing type.